engine/core/engine.cpp

```diff
@@ -317,14 +293,14 @@ void Engine::setSettingsLoaderDefaults() {
     Engine::settingsLoader->save();
 }
 
-Window* Engine::getWindow() {
+Frame* Engine::getWindow() {
```

Comment:
`Frame` isn't intended to be used like a window, could you move the windowing logic into an abstract Window class and then subclass it?